adaption_id->adaption_type in endpoints and as syslog foreign key

fredriksvendsen opened the pull request:
+ some formatting as requested

jfremstad committed:
Log by adaption-type

jfremstad committed:
JOIN syslog and adaption in filter

skykanin committed:
Added table for network-collections

skykanin committed:
Added table for network-collections

skykanin committed:
Added functions for inserting and querying the new networkcollections-table

skykanin committed:
Added functions for inserting and querying the new networkcollections-table

skykanin committed:
Added http get handler for adaption requests

skykanin committed:
Added http get handler for adaption requests

skykanin committed:
Fixed syntax error in ns require for adaption handler file

skykanin committed:
Fixed syntax error in ns require for adaption handler file

skykanin committed:
Added adaption handler to route

skykanin committed:
Added adaption handler to route

skykanin committed:
Backend test now pass again

skykanin committed:
Backend test now pass again

allmeta committed:
-changed fetch function to return promise
-gets adaption and device ids for filtering from database

allmeta committed:
made adaption type filter option

allmeta committed:
correctly maps over headers from first fetched res

allmeta committed:
added live update

allmeta committed:
bit better live update button

fredriksvendsen committed:
changed from adaption_id to adaption_type in endpoints and as foreign key in db. + format as requested

fredriksvendsen committed:
changed from adaption_id to adaption_type in endpoints and as foreign key in db. + format as requested

Commit:
Added Mapnodes to NetColl in mocker

Commit:
Added Mapnodes to NetColl in mocker

Commit:
Added Mapnodes to required values in the mocker

Commit:
Added Mapnodes to required values in the mocker

skykanin committed:
Finished response object

skykanin committed:
Finished response object

skykanin committed:
Merge branch '107-adaption-request' of github.com:it2901/Skrp into 107-adaption-request

skykanin committed:
Command builder gets correct args passed to it

YngveMolnes commented on database/001_initial_schema.sql:
```diff
@@ -21,7 +21,7 @@ CREATE TABLE IF NOT EXISTS config (
 CREATE TABLE IF NOT EXISTS public.adaption (
   adaption_id SERIAL PRIMARY KEY NOT NULL,
-  adaption_type TEXT NOT NULL,
+  adaption_type TEXT NOT NULL UNIQUE,
```
These changes does not make sense. Instead you should leave the data model as is. If you want to get adaption type in the with the log data, you can join this in.

SELECT foo, bar FROM system_log NATURAL JOIN adaption;

YngveMolnes replied:
It will also break the current datadump, so if you want to make this change, the datadump should be revised.

YngveMolnes replied:
Just ask me with what you are trying to solve, and i can help you with the database stuff (-:

YngveMolnes replied:
and also, adaption_type is not meant to be unique, think of it as an enum. We should probably have changed this to a while a go, but that would limit us from adding additional types without changing the model every time.

YngveMolnes commented on database/001_initial_schema.sql:
```diff
@@ -31,7 +31,7 @@ CREATE TABLE IF NOT EXISTS public.system_log (
   system_log_id SERIAL PRIMARY KEY NOT NULL,
   device_id INTEGER NOT NULL REFERENCES device(device_id),
-  adaption_id INTEGER REFERENCES adaption(adaption_id),
+  adaption_type TEXT NOT NULL REFERENCES adaption(adaption_type),
```
See above comments

Commit:
updated mocker to give out raw NetJSON data

Commit:
updated mocker to give out raw NetJSON data

Commit:
updated the map and nodegraph to match the new mocker endpoints

Commit:
updated the map and nodegraph to match the new mocker endpoints

skykanin committed:
Broken stuff

skykanin committed:
Fixed formatting

skykanin committed:
Added more test cases

skykanin committed:
Added logging to endpoint, so that all incoming netcolls are logged

skykanin committed:
Updated test to mock out db logging from handler

skykanin committed:
Fixed bugs in db queries

Commit:
removed netcoll, updated all NetJSON objects to be wrapped in Netcollection, made a main of all NetJSON Objects

Commit:
updated the last of the schemas

Commit:
Merge branch 'develop' into 109-log-final

Commit:
Merge pull request #111 from it2901/109-log-final

109 log final

Commit:
Merge branch 'develop' into updating-mocker-to-give-valid-data

Commit:
Added field to Join on between geoloc and networkgph

Commit:
Merge pull request #113 from it2901/updating-mocker-to-give-valid-data

Updating mocker to give valid data

Commit:
Bug fix 

Removed label from required value in mocker, to allow unique IDs for Nodes in the network graph"

skykanin committed:
Route now is a POST and accepts a JSON body

skykanin committed:
Changed name of route

skykanin committed:
Updated test for collection post route

skykanin committed:
Fixed code docs

skykanin committed:
Fixed foramtting of unit test

skykanin committed:
Merge branch 'develop' into 107-adaption-request

skykanin committed:
Merge pull request #108 from it2901/107-adaption-request

107 adaption request endpoint

skykanin committed:
Fixed string error

fredriksvendsen committed:
adaption_id -> _type, requested changes++

fredriksvendsen committed:
resolving merge conflict
